frontend/public/components/marketplace/kubernetes-marketplace-items.jsx

```diff
@@ -0,0 +1,307 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import * as PropTypes from 'prop-types';
+import {CatalogTileView} from 'patternfly-react-extensions/dist/esm/components/CatalogTileView';
+import {CatalogTile} from 'patternfly-react-extensions/dist/esm/components/CatalogTile';
+import {FilterSidePanel} from 'patternfly-react-extensions/dist/esm/components/FilterSidePanel';
+import {EmptyState} from 'patternfly-react/dist/esm/components/EmptyState';
+import FormControl from 'patternfly-react/dist/esm/components/Form/FormControl';
+
+import {normalizeIconClass} from '../catalog-item-icon';
+
+const getFilters = (items, filterCategories=[]) => {
+  const filters = {
+    byName: {
+      value: '',
+      active: 'false',
+    },
+  };
+
+  for (let i=0; i<items.length; i++){
+    const item = items[i];
+    for (const key in item) {
+      if (!item.hasOwnProperty(key) || key === 'obj' || key.includes('IconClass') || key.includes('ImgUrl')) {
+        continue;
+      }
+      // Only consider filters in filterCategories
+      if (filterCategories.length > 0 && !filterCategories.includes(key.replace(/tile/, '').toLowerCase())) {
+        continue;
+      }
+      if (item[key] && (!filters[`by${key}`] || !filters[`by${key}`][item[key]]) ) {
+        let filterKey = key;
```

Comment:
Not sure I understand why you need a copy of key here. Did you mean to do something like

```javascript
let filterKey = `by${key}`;
```

Reply:
That's exactly what I meant to do. I'll define `filterKey` outside the if statement so I can define it in one spot.